Form

gd-hxy opened the pull request:
添加页面已完成

gd-hxy committed:
1、 修改了类型选择后出现的tips信息
2、 取消了可选择的联系方式，用户只能填写手机号码（删除了QQ和微信）
3、 完成了图片上传、预览、删除的js代码

gd-hxy committed:
1、 增加了在用户上传5张图片后隐藏上传按钮的功能
2、 实现了图片的横向排列

gd-hxy committed:
完成了表单验证

gd-hxy committed:
完成了toast组件的引入，完成了代码发布功能

cccRaim committed:
toast美化&mock修复

gd-hxy committed:
1、改进了tip的显示方式
2、修改了表单验证的方式，从填完所有信息发布按钮才亮起改为按下发布按钮后再验证

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
+      if (this.title === '') {
+        this.showToast('标题不能为空')
+        console.log('title wrong')
+        this.titleState = false
```
console.log要去掉，titleState是什么，为什么要有这个变量？

cccRaim replied:
其他地方的也同理

gd-hxy replied:
已修改

cccRaim replied:
修改好后就 commit，评论会自动消失的

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
+        {type: '雨伞', index: 2},
+        {type: '钥匙', index: 3},
+        {type: '图书', index: 4},
+        {type: '其它', index: 5}
```
item的数据最好从服务端获取

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
+        {type: '其它', index: 5}
+      ],
+      selected: '',
+      Else: '',
```
我看了一下没有地方用到Else，还是首字母大写的，去除掉没用的变量吧

gd-hxy replied:
已去除

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
+      place: '',
+      contact: '',
+      limit: 5,
+      size: 0,
```
为什么要有size这个变量，直接this.imgLIst.length不就好了

gd-hxy replied:
下面有其他的判断要用到这个变量，不能单单只用this.ingList.length

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
   data () {
     return {
+      isActive: true,
+      button: 'u-button',
```
这个变量不应该维护class，如果非要加个变量，应该是表单是否合理，比如isValidated，然后template那边再通过isValidated来控制class选哪个

gd-hxy replied:
已修改

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
   },
   data () {
     return {
+      isActive: true,
```
这个变量没用到

gd-hxy replied:
已去除

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
+      placeState: false,
+      contactState: false
     }
   },
```
其实为了提交数据更清晰，你可以把表单相关的变量全部维护到一个form的object里，然后所有要提交的数据都存在form这个变量里，这样更清晰

gd-hxy replied:
已修改

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
+    ]),
+    // image upload
+    chooseType () {
+      document.getElementById('upload_file').click()
```
不要有dom操作

gd-hxy replied:
已修改

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -77,35 +77,285 @@
+    fileChange (el) {
+      if (!el.target.files[0].size) return
+      this.fileList(el.target)
+      el.target.value = ''
```
不要有dom操作

gd-hxy committed:
1、删除了多余变量
2、删除了表单验证时的console.log操作
3、将文件上传时的dom操作改为了vue操作
4、定义了数据渲染方法updataAddItemList
5、在main与store中定义了该方法

gd-hxy committed:
mapState中将find修改为add

cccRaim commented on src/pages/add/index.vue:
```diff
@@ -45,12 +47,9 @@
-                <option value="4">校园卡3</option>
+              <select v-model="selected">
+                <option>请选择</option>
+                <option v-for="item in items" :key="item.index" :source="itemList">{{item.type}}</option>
```
确定是 source？

cccRaim replied:
确定是 source？

cccRaim replied:
确定是 source？

cccRaim replied:
确定是 source？

cccRaim replied:
不应该是 srouce 吧

gd-hxy committed:
修改了itemList渲染错误

cccRaim committed:
Merge branch 'master' into Form

cccRaim reviewed (approved):
感觉可以了，虽然还有一些小细节，不过以后联调的时候再说
